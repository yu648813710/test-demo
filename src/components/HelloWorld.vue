<template>
  <div class="hello">
    <p>前列英雄</p>
    <div class="list_hero" >
      <div class="item" v-for="(item,index) in hero" v-show="index<4" :key="'hero'+index" @click="goDetail(item.id)">{{item.name}}</div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'HelloWorld',
  props: {
    hero: Array
  },
  methods:{
     goDetail(e){
          this.$emit("goDetail",e);
      }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
p{
  font-size:28px; 
}
.list_hero{
  overflow: hidden;
  height: auto;
  display: flex;
  justify-content: space-around;
  .item{
    width: 150px;
    height: 100px;
    background: #eaeaea;
    line-height: 100px;
    cursor: pointer;
    &:hover{
      background: #42b983;
      color:#fff;
    }
  }
}
</style>
