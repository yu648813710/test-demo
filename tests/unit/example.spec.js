import {
  shallowMount,
  mount
} from '@vue/test-utils'
import HelloWorld from '@/components/HelloWorld.vue'

describe("测试hello组件", () => {
  const wrapper = mount(HelloWorld, {
    propsData: {
      hero: [{
        id: 1,
        name: "哈哈"
      }]
    }
  });
  test('测试hello组件的dome节点渲染', () => {
    
    expect(wrapper.html()).toContain("<div class=\"hello\"><p>前列英雄</p> <div class=\"list_hero\"><div class=\"item\">哈哈</div></div></div>");
  });
  test("测试hello组件的点击事件", () => {
      wrapper.find('div.item').trigger('click');
      expect(wrapper.emitted().goDetail).toEqual([[1]]);
  })
});