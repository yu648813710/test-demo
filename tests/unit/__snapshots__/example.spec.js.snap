// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  class="hello"
>
  <p>
    前列英雄
  </p>
   
  <div
    class="list_hero"
  />
</div>
`;

exports[`测试hello组件 1`] = `
<div
  class="hello"
>
  <p>
    前列英雄
  </p>
   
  <div
    class="list_hero"
  >
    <div
      class="item"
    >
      哈哈
    </div>
  </div>
</div>
`;

exports[`测试hello组件 测试hello组件的dome节点渲染 1`] = `
<div
  class="hello"
>
  <p>
    前列英雄
  </p>
   
  <div
    class="list_hero"
  >
    <div
      class="item"
    >
      哈哈
    </div>
  </div>
</div>
`;
